### PR TITLE
Skip balancer api repository when url is not defined

### PR DESCRIPTION
**Pull request by agualis**
# Description

When connected to Goerli, you can see the attached error when loading the index page.

The problem is that [goerli.json](https://github.com/balancer-labs/frontend-v2/blob/870d008f7062e3082cb446c895a8df61f4bab8ba/src/lib/config/goerli.json) does not have an attribute for balancerApi so that the SDK.

This solution will also work for `docker.json`.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [x] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Visual context

Please provide any relevant visual context for UI changes or additions. This could be static screenshots or a loom screencast.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] If package-lock.json has changes, it was intentional.
- [ ] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

**Comment by timjrobinson** [comment]
@agualis I like the style updates. The balancer API is also used in usePoolsQuery so I updated it there too. I renamed + moved the isBalancerApiDefined check to a util function so it can be shared between them.

Tested this on goerli + mainnet and everything appears to work correctly. Noticed that console.log appears 5 times per individual pool page because usePoolQuery is called from so many different paths, this is something we can optimise in the future.

**Review by timjrobinson — APPROVED** [review]
If you think my changes are good and work for you feel free to merge. Your code LGTM.

**Comment by agualis** [comment]
@timjrobinson Thanks Tim, your changes look good. 

I also noticed the 5X console.log and thought that it would remind us that we can do some optimisations in the future 👀

Just one small detail about the naming:  I did not want to use a "boolean function name" like `isBalancerApiDefined` because the function returns a boolean but also contains a side effect (console.log()). We can refactor following the advice in[ this post](https://www.teamten.com/lawrence/programming/keep-if-clauses-side-effect-free.html) or we can also keep it like this cause the side effect is not so important in this scenario. I'm ok with both decisions.

File: src/lib/utils/balancer/api.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { configService } from '@/services/config/config.service';
+
+export function isBalancerApiDefined() {
+  const defined = configService.network.balancerApi;
+  if (!defined)
+    console.log(
+      `Skipping balancer api provider in your current network (${configService.network.chainName})`
+    );
+  return defined;
+}
```
Comment (garethfuller) [inline src/lib/utils/balancer/api.ts:10]: I don't think we need to log here and this should return a boolean.
```suggestion
export const isBalancerApiDefined = !!configService.network.balancerApi
```

File: src/composables/queries/usePoolsQuery.ts [hunk 2]
```diff
@@ -104,6 +101,18 @@ export default function usePoolsQuery(
...
+    const providers: SDKPoolRepository[] = [];
+    if (isBalancerApiDefined()) {
+      const balancerApiRepository = initializeDecoratedAPIRepository();
+      providers.push(balancerApiRepository);
+    }
+    const subgraphRepository = initializeDecoratedSubgraphRepository();
+    providers.push(subgraphRepository);
+
+    return providers;
+  }
```
Comment (garethfuller) [inline src/composables/queries/usePoolsQuery.ts:114]: Lets stick to the naming convention already established.
```suggestion
  function buildRepositories() {
    const repositories: SDKPoolRepository[] = [];
    if (isBalancerApiDefined()) {
      const balancerApiRepository = initializeDecoratedAPIRepository();
      repositories.push(balancerApiRepository);
    }
    const subgraphRepository = initializeDecoratedSubgraphRepository();
    repositories.push(subgraphRepository);

    return repositories;
  }
```

File: src/services/pool/pool.repository.ts [hunk 3]
```diff
@@ -57,4 +57,16 @@ export default class PoolRepository {
...
+    const providers: SDKPoolRepository[] = [];
+    if (isBalancerApiDefined()) {
+      const balancerApiRepository = this.initializeDecoratedAPIRepository();
+      providers.push(balancerApiRepository);
+    }
+    const subgraphRepository = this.initializeDecoratedSubgraphRepository();
+    providers.push(subgraphRepository);
+
+    return providers;
+  }
```
Comment (garethfuller) [inline src/services/pool/pool.repository.ts:71]: Can we try to define this function in one place?

**Review by garethfuller — CHANGES_REQUESTED** [review]

File: src/services/pool/pool.repository.ts (hunk 3 above)
Comment (agualis) [reply src/services/pool/pool.repository.ts]: @garethfuller It is not trivial so I suggest that @timjrobinson or me try to do it later in a separate PR.

Comment (garethfuller) [reply src/services/pool/pool.repository.ts]: Can usePoolsQuery not use `PoolRepository`? Is it doing something different?

Comment (agualis) [reply src/services/pool/pool.repository.ts]: The current code shows that `initializeDecoratedAPIRepository` and `initializeDecoratedSubgraphRepository` are slightly different. Probably refactoring to use PoolRepository in both places would avoid that duplication but I'm not so familiar with this code and that's why I check it carefully check it in a different PR.

Comment (timjrobinson) [reply src/services/pool/pool.repository.ts]: Yea I'd rather just get this merged and do that refactor later.

**Review by timjrobinson — APPROVED** [review]

File: src/services/pool/pool.repository.ts (hunk 3 above)
Comment (agualis) [reply src/services/pool/pool.repository.ts]: @garethfuller If you agree, could you approve this PR so that we can merge?

**Review by garethfuller — APPROVED** [review]

**Pull request merged**